proposals/0000-split-wunused-imports.md

```diff
@@ -0,0 +1,230 @@
+    * warnDuplicateImports: import M + import M(f), even when f is used complain about duplicate import of f
+1. `-Wall` includes `-Wunused-imports`
+
+Proposed:
+
+0. `-Wunused-imports` reports
+    * warnUnusedModules: import M, where nothing is used from M
+    * warnUnusedImports: import M(f), where f is unused, and M doesn't fall under warnUnusedModules
+1. `-Wduplicate-imports`
+    * warnDuplicateImports: import M + import M(f), even when f is used complain about duplicate import of f
+2. `-Wall` includes `-Wunused-imports`, but **not** `-Wduplicate-imports`
+
+### Meta warning
+
+One big drawback of the proposed changed is that, as is, the default behaviour will change silently - the user was not asked and has not agreed for this to change.
+
+Therefore, we propose that we introduce a "meta warning", let's say `-Wmeta-unused-imports` (names here are placeholders for now), with the following properties:
+
+0. Whenever a user has `-Wunused-imports` on, and they don't have `-Wno-meta-unused-imports` on, emit a warning that highlights the changed behaviour of `-Wunused-imports`.
+1. It is on by default.
+2. It **is not made into an error by `-Werror`**. Optionally there could be `-Wmeta-error` or something along those lines for users who want to get errors for potential future meta warnings.
+
+This `-Wmeta-unused-improts` warning is only present for a few release cycles, and is removed afterwards, or alternatively, turned off by default.
+
+## Examples
+
+###
+
+```haskell
+import Foo
+import Foo (x)
+
+bla = x
+```
+
+Current:
+* with `-Wunused-imports` - warn that the `Foo` import is unused
+
+Proposed:
+* with `-Wunused-imports` - nothing
+* with `-Wduplicate-imports` - warn that the `Foo` import is duplicate
+
+###
+
+```haskell
+import Foo (x)
+import Bar (x)
+
+bla = x
+```
+
+Current:
+* with `-Wunused-imports` - warn that the `Bar` import is unused
+
+Proposed:
+* with `-Wunused-imports` - nothing
+* with `-Wduplicate-imports` - warn that the `Bar` import is duplicate
+
+###
+
+```haskell
+import Foo
+import Bar
+
+bla = x
+```
+
+Current:
+* with `-Wunused-imports` - warn that the `Bar` import is unused
+
+Proposed:
+* with `-Wunused-imports` - nothing
+* with `-Wduplicate-imports` - warn that the `Bar` import is duplicate
+
+###
+
+```haskell
+import Foo
+
+bla = ()
+```
+
+Current:
+* with `-Wunused-imports` - warn that the `Foo` import is unused
+
+Proposed:
+* with `-Wunused-imports` - warn that the `Foo` import is unused
+* with `-Wduplicate-imports` - nothing
+
+###
+
+```haskell
+import Foo (x)
+
+bla = ()
+```
+
+Current:
+* with `-Wunused-imports` - warn that the `Foo` import is unused
+
+Proposed:
+* with `-Wunused-imports` - warn that the `Foo` import is unused
+* with `-Wduplicate-imports` - nothing
+
+## Effect and Interactions
+
+Unsure what to fill in here, it seems that the [Proposed Change Specification](#proposed-change-specification) covers the effects of this change.
+
+## Costs and Drawbacks
```

Comment:
An obscure backwards compatibility point: with this proposal, compiling a module with duplicate imports will fail under `-Werror -Weverything -Wno-unused-imports`, whereas previously it would have succeeded (since `-Wno-unused-imports` previously suppressed both).

We could avoid this by making `-Wunused-imports` into a group that includes both `-Wreally-unused-imports` (in `-Wall`) and `-Wduplicate-imports` (in `-Weverything`). This would also mean that users who explicitly ask for `-Wunused-imports` continue to get both.

Is this worth it? I'm not sure. Adding a group feels a bit fiddly for a comparatively rare edge case.